transport/rif_comms/node.py

```diff
@@ -1,51 +1,339 @@
+
+        self._stop_event = Event()
+        self._stop_event.set()
+
+        self.log = log.bind(node_address=pex(self.address))
+
+    @property
+    def _queueids_to_queues(self) -> QueueIdsToQueues:
+        chain_state = views.state_from_raiden(self._raiden_service)
+        return views.get_all_messagequeues(chain_state)
 
     def start(self, raiden_service: RaidenService, message_handler: MessageHandler, prev_auth_data: str):
-        raise NotImplementedError
+        if not self._stop_event.ready():
+            raise RuntimeError(f"{self!r} already started")
+        self._stop_event.clear()
+        self._raiden_service = raiden_service
+
+        self._rif_comms_connect_stream = self._client.connect()
+        self._our_topic = self._client.subscribe(to_checksum_address(raiden_service.address))
+        self._client.get_peer_id(to_checksum_address(raiden_service.address)) # TODO remove this after grpc api request blocking is fixed
+        for message_queue in self._address_to_message_queue.values():
+            if not message_queue:
+                self.log.debug("Starting message_queue", message_queue=message_queue)
+                message_queue.start()
+
+        self.start_listener_thread()
+        self._greenlets = [self._our_topic_thread]
+        self.log.debug("RIF Comms Node started", config=self._config)
+        Runnable.start(self)
+
+    def __repr__(self):
+        if self._raiden_service is not None:
+            node = f" node:{pex(self._raiden_service.address)}"
+        else:
+            node = ""
+
+        return f"<{self.__class__.__name__}{node} id:{id(self)}>"
+
+    def _run(self) -> None:
+        """ Runnable main method, perform wait on long-running subtasks """
+        # dispatch auth data on first scheduling after start
+        self.greenlet.name = f"RifCommsNode._run node:{pex(self._raiden_service.address)}"
+        try:
+            # waits on _stop_event.ready()
+            # children crashes should throw an exception here
+            self.log.info("RIF Comms _run")
+        except GreenletExit:  # killed without exception
+            self._stop_event.set()
+            killall(self.greenlets)  # kill children
+            raise  # re-raise to keep killed status
+        except Exception:
+            self.stop()  # ensure cleanup and wait on subtasks
+            raise
 
     def stop(self):
-        raise NotImplementedError
+        """
+        Try to gracefully stop the greenlet synchronously
+
+        Stop isn't expected to re-raise greenlet _run exception
+        (use self.greenlet.get() for that),
+        but it should raise any stop-time exception
+
+        Disconnects from RIF Communications node
+        """
+        if self._stop_event.ready():
+            return
+        self._stop_event.set()
+
+        for message_queue in self._address_to_message_queue.values():
+            if message_queue:  # if message_queue.greenlet is not None
```

Comment:
```suggestion
            if message_queue.greenlet:
```

imo, going all the way from `RifCommsNode` to `Runnable` to `__bool__` is too obscure a path for this line to make sense to the reader

Reply:
I basically put this comment here for that. I found usable to remember that, but i can remove it.